gc.c

```diff
@@ -12168,6 +12170,8 @@ objspace_malloc_prepare(rb_objspace_t *objspace, size_t size)
 static inline void *
 objspace_malloc_fixup(rb_objspace_t *objspace, void *mem, size_t size)
 {
+    if (UNLIKELY(during_gc && !rb_multi_ractor_p() && ruby_thread_has_gvl_p())) rb_bug("Cannot malloc during GC");
```

Comment:
> `!rb_multi_ractor_p() && ruby_thread_has_gvl_p())`

why checking them?

Reply:
I added a comment to explain.

We need `!rb_multi_ractor_p()` because ractors could run while another thread is sweeping.

We need `ruby_thread_has_gvl_p()` because if this thread doesn't have the GVL, then it will try to acquire the GVL and wait for GC to finish before starting GC.